### PR TITLE
Remove ar_debug requirement for cookie-based debug reporting

### LINKED ISSUE #1440: Remove the ar_debug requirement for cookie-based debug reporting while keeping the requirement for third-party cookie availability
Currently Attribution Reporting API cookie-based debug reporting is allowed when third-party cookies are available AND the special unpartitioned ar_debug cookie is set.

To make it easier for reporting origins to set up cookie-based debug reporting, we propose to keep the requirement of third-party cookies accessibility for the reporting origin on the source/destination sites, but remove the ar_debug cookie requirement. 

There is no significant difference from either privacy or utility perspectives.
- Ad-techs can only set the ar_debug cookie if they have access to third-party cookies on the source/destination sites. Therefore, third-party cookie accessibility is equivalent to the ad-techs’ ability to set the ar_debug cookie, and thus there is no privacy regression.
- From the utility perspective, reporting origins can still control whether to enable cookie-based debug reporting via `debug_key`([attribution-success debug reports](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#attribution-success-debugging-reports)) or `debug_reporting`([verbose debug reporting](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#verbose-debugging-reports)) fields, and thus there is no utility regression.

This proposal brings ARA debugging more aligned with PAA which doesn’t require any API-specific cookie to be set. There may also be a slight utility improvement where ARA debug reports are not received due to misconfigured cookies ([example](https://github.com/WICG/attribution-reporting-api/issues/1195)).

### PR DESCRIPTION
**Pull request by linnan-github**
Fixes #1440

***
<a href="https://pr-preview.s3.amazonaws.com/linnan-github/conversion-measurement-api/pull/1441.html" title="Last updated on Oct 15, 2024, 12:55 PM UTC (1ec425a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/1441/2877fda...linnan-github:1ec425a.html" title="Last updated on Oct 15, 2024, 12:55 PM UTC (1ec425a)">Diff</a>

### CODE REVIEW

File: EVENT.md [hunk 1]
```diff
@@ -725,28 +725,14 @@ fully understood during roll-out and help flush out any bugs (either in browser
...
-available in a transitional phase while third-party cookies are available and
-are already capable of user tracking, the browser will check (at both source
-and trigger registration) for the presence of a special cookie
-set by the reporting origin:
-```http
-Set-Cookie: ar_debug=1; SameSite=None; Secure; HttpOnly
-```
-If a cookie of this form is not present, debugging information will be ignored. Additionally,
-browsers may choose to enable debugging for specific use-cases (for example, reporting origins
+The debugging information will only be available while third-party cookies are
```
Comment (akashnadan) [inline EVENT.md:728]:
```suggestion
The debugging information will only be available if third-party cookies are
```

File: EVENT.md [hunk 2]
```diff
@@ -725,28 +725,14 @@ fully understood during roll-out and help flush out any bugs (either in browser
...
+available and are already capable of user tracking, and will be ignored if
```
Comment (akashnadan) [inline EVENT.md:729]:
```suggestion
available for current site, and will be ignored if
```

File: EVENT.md [hunk 3]
```diff
@@ -725,28 +725,14 @@ fully understood during roll-out and help flush out any bugs (either in browser
...
+third-party cookies are disabled/deprecated generally or for a particular site.
```
Comment (akashnadan) [inline EVENT.md:730]:
```suggestion
third-party cookies are disabled generally or for a particular site.
```

**Review by akashnadan — CHANGES_REQUESTED** [review]
Minor edits

**Review by johnivdel — APPROVED** [review]
lgtm

File: EVENT.md [hunk 4]
```diff
@@ -725,28 +725,14 @@ fully understood during roll-out and help flush out any bugs (either in browser
...
-are already capable of user tracking, the browser will check (at both source
-and trigger registration) for the presence of a special cookie
-set by the reporting origin:
-```http
-Set-Cookie: ar_debug=1; SameSite=None; Secure; HttpOnly
-```
-If a cookie of this form is not present, debugging information will be ignored. Additionally,
-browsers may choose to enable debugging for specific use-cases (for example, reporting origins
+The debugging information will only be available if third-party cookies are
+available for current site, and will be ignored if
```
Comment (apasel422) [inline EVENT.md:729]:
```suggestion
available for the current site, and will be ignored if
```

File: EVENT.md [hunk 5]
```diff
@@ -818,8 +804,11 @@ The debugging reports will be sent to a new endpoint:
 https://<reporting origin>/.well-known/attribution-reporting/debug/verbose
 ```
 
-In order to receive verbose debug reports on trigger registrations, the special
-`ar_debug` cookie needs to be present for both source and trigger registrations.
+In order to receive verbose debug reports on trigger registrations, the
+reporting origin needs to be able to access third-party cookies on the
+destination site. If the trigger is attributed to a source, the reporting
+origin also needs to be able to access third-party cookies on the source site
```
Comment (apasel422) [inline EVENT.md:810]:
```suggestion
origin also needed to be able to access third-party cookies on the source site
```

**Review by apasel422 — APPROVED** [review]

File: EVENT.md [hunk 6]
```diff
@@ -725,28 +725,14 @@ fully understood during roll-out and help flush out any bugs (either in browser
...
-and trigger registration) for the presence of a special cookie
-set by the reporting origin:
-```http
-Set-Cookie: ar_debug=1; SameSite=None; Secure; HttpOnly
-```
-If a cookie of this form is not present, debugging information will be ignored. Additionally,
-browsers may choose to enable debugging for specific use-cases (for example, reporting origins
+The debugging information will only be available if third-party cookies are
+available for the current site, and will be ignored if
+third-party cookies are disabled generally or for a particular site.
```
Comment (apasel422) [inline EVENT.md:730]:
```suggestion
Debugging will only be enabled if third-party cookies are
available for the current site, and will be disabled if
third-party cookies are disabled generally or for a particular site.
```

**Review by apasel422 — APPROVED** [review]

**Pull request merged**